### ISSUE TITLE
clone3 E2E don't copy the data I insert

### ISSUE THREAD

**Issue by muyingluo1997** [p4lang/behavioral-model#1075]
I have used `clone3(CloneType.E2E,(bit<32>)32w250,{standard_metadata});` in Egress Pipeline. But I found the bmv2 didin't clone the header and data that I insert in the packet, I don't really know is that the expression of clone3 I used is wrong?
Could anyone give me some tips? 

Example:
I insert some data in switch-1. When the packet routing to switch-2, I used  `clone3(CloneType.E2E,(bit<32>)32w250,{standard_metadata});`, and found the data I inserted in switch-1 was disappeared.

**Comment by jafingerhut** [comment]
I would recommend asking questions about P4 programming and debugging on https://forum.p4.org. This github repo is for reporting bugs and/or enhancement requests to the bmv2 software switch, not P4 programming questions.

If you do want help, it helps others figure out what you are trying to do if you can give more detail than you have in your question above.  For example, if you have published your P4 code, the network topology you run it on, what table entries are installed that are relevant, what packet you sent in to a switch, what packet came out, and how it differs from what you expected -- in general, if there is enough detail to reproduce the behavior you are seeing, it is more likely someone can diagnose the reason for the behavior.

**Comment by muyingluo1997** [comment]
I have solved this problem. I thought that when I used clone3 in a line of code, the data packet of the clone was what the data packet looked like when it went to the current code.
Then I noticed a sentence mentioned in (https://git.rnp.br/cnar/sdn-multicamada/onos/blob/7bc181fe9b5642841fbdef166ae82bd36ee4bc2c/pipelines/main/src/main/resources/tofino-mavericks/main.p4pp).
` * Note 2: There are ongoing discussions among P4 working group
 * members in 2019-Apr regarding exactly how resubmit, recirculate,
 * and clone3 operations can be called anywhere in their respective
 * controls, but the values of the fields to be preserved is the value
 * they have when that control is finished executing.  That is how
 * these operations behave in P4_14, but this requires some care in
 * making this happen in P4_16.
`
In my p4 code, I use clone3 in egress and use setInvalid() at the end to clear the previously collected fields. Therefore, I can't see the data collected before.
